scripts/http-form-fuzzer.nse

```diff
@@ -201,15 +234,22 @@ function action(host, port)
       for _,form_plain in ipairs(all_forms) do
         local form = http.parse_form(form_plain)
         if form and form.action then
-          local affected_fields = fuzz_form(form, minlen, maxlen, host, port, path)
+          local affected_fields = fuzz_form(form, minlen, maxlen, timeLimit, end_time, host, port, path)
           if #affected_fields > 0 then
             affected_fields["name"] = "Path: "..path.." Action: "..form["action"]
             table.insert(return_table, affected_fields)
           end
         end
+        if (timeLimit and nmap.clock_ms() > end_time) then
```

Comment:
Just like we did with `fuzz_field`, have `fuzz_form` return a value that indicates it timed out. Otherwise maybe the script finished just before the time limit and we only marked it timed out because we checked too late. There should really only be one comparison with `nmap.clock_ms()` in the whole script: in the innermost function/loop.